contracts/src/token/erc20/extensions/flash_mint.rs

```diff
@@ -0,0 +1,397 @@
+    /// it's used in a token with a cap like
+    /// [`crate::token::erc20::extensions::Capped`], make sure to override this
+    /// function to integrate the cap instead of [`U256::MAX`].
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token` - The address of the token that is requested.
+    /// * `erc20` - Read access to an [`Erc20`] contract.
+    fn max_flash_loan(&self, token: Address, erc20: &Erc20) -> U256;
+
+    /// Returns the fee applied when doing flash loans.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token` - The token to be flash loaned.
+    /// * `amount` - The amount of tokens to be loaned.
+    ///
+    /// # Errors
+    ///
+    /// If the token is not supported, then the error
+    /// [`Error::UnsupportedToken`] is returned.
+    fn flash_fee(
+        &self,
+        token: Address,
+        amount: U256,
+    ) -> Result<U256, Self::Error>;
+
+    /// Performs a flash loan.
+    ///
+    /// New tokens are minted and sent to the `receiver`, who is required to
+    /// implement the [`IERC3156FlashBorrower`] interface. By the end of the
+    /// flash loan, the receiver is expected to own value + fee tokens and have
+    /// them approved back to the token contract itself so they can be burned.
+    ///
+    /// Returns a boolean value indicating whether the operation succeeded.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `receiver` - The receiver of the flash loan. Should implement the
+    ///   [`IERC3156FlashBorrower::on_flash_loan`] interface.
+    /// * `token` - The token to be flash loaned. Only [`contract::address()`]
+    ///   is supported.
+    /// * `value` - The amount of tokens to be loaned.
+    /// * `data` - Arbitrary data that is passed to the receiver.
+    /// * `erc20` - Write access to an [`Erc20`] contract.
+    ///
+    /// # Errors
+    ///
+    /// If the `amount` is greater than the value returned by
+    /// [`IErc3156FlashLender::max_flash_loan`], then the error
+    /// [`Error::ExceededMaxLoan`] is returned. If `token` is not supported,
+    /// then the error [`Error::UnsupportedToken`] is returned.
+    /// If the `token` address is not a contract, then the error
+    /// [`Error::InvalidReceiver`] is returned. If the contract fails to
+    /// execute the call, then the error [`Error::InvalidReceiver`] is returned.
+    /// If the receiver does not return [`BORROWER_CALLBACK_VALUE`], then the
+    /// error [`Error::InvalidReceiver`] is returned.
+    ///
+    /// # Events
+    ///
+    /// Emits an [`erc20::Transfer`] event.
+    /// Emits an [`erc20::Approval`] event.
+    ///
+    /// # Panics
+    ///
+    /// If the new (temporary) total supply exceeds `U256::MAX`.
+    /// If the sum of the loan amount and fee exceeds the maximum value of
+    /// `U256::MAX`.
+    fn flash_loan(
```

Comment:
Add a note about making the function callable, like we have [here](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/main/contracts/src/token/erc721/extensions/metadata.rs#L93-L102).

Reply:
Maybe we should put the "NOTE: in order to have..." part _below_ the `# Examples` header? 
We wouldn't even need the "NOTE:" part.

Current function hover tooltip (examples above the header):
![image](https://github.com/user-attachments/assets/7268e838-8ed7-4929-98c9-87512cd09829)

If below:
![image](https://github.com/user-attachments/assets/58455b06-954a-4910-a814-7010fe80cf07)

@bidzyyys @qalisander

Reply:
I don't think it should be a part of `Examples`. It is a "must" recommendation currently.
I feel this note should be before any paragraph. Right after general description

Reply:
That makes sense too. We should then either remove the `# Examples` header OR make it plain test, like:
```rust
/// Example:
///
/// ```rust,ignore
/// fn flash_loan...
/// ...
```